wordpress-mobile/WordPress-Android
# Issue/11153 insert multiple edited images into gutenberg

Partially fixes #11153

Merge instructions:
1. Review the code 
2. Make sure the issues in the `notes` section are fixed
3. Test the changes
4. Make sure https://github.com/wordpress-mobile/WordPress-Android/pull/11686 is merged
5. Update target branch to `issue/11153-update-image-thumbnail-after-crop`
6. Merge the PR


Notes:
The feature still has a vital issues and the flow doesn't fully work yet. However, I didn't want to make even more changes to this PR. I think the best approach is to fix the issue in another PRs branched from this PR.
- [ ] When the PreviewImage flow is started with images saved in local storage, Glide's `asFile()` method returns an error - nothing happens after you click on "Crop" action in the Toolbar. We probably shouldn't even call `asFile()` on local images, wdyt? Moreover, we probably should show a snackbar and show an error to the user when `asFile` fails.

To test:
- Test after the issue mentioned above is fixed

1. Add an ImageBlock
2. Select "Add an image from local storage"
3. Select an image and click on "Edit"
4. Crop the image
5. Notice the cropped image gets added to the imageBlock
-------------------------------
1. Add a Gallery block
2. Select "Add an image from local storage"
3. Select multiple images and click on "Edit"
4. Crop some of the images
5. Click on "Insert"
6. Verify all images are added and the cropped images are actually cropped

PR submission checklist:

- [] I have considered adding accessibility improvements for my changes.
